### PR TITLE
CLC-8002, Ripley gets schema.sql

### PR DESCRIPTION
**Pull request by johncrossman**
https://jira-secure.berkeley.edu/browse/CLC-8002

Time to fire up Nostromo.

image

For comparison, here is [the old schema script](https://github.com/ets-berkeley-edu/junction/blob/master/db/schema.rb).  I have excluded `oec_course_codes` and `webcast_course_site_log`.

### CODE REVIEW

**Comment by johncrossman** [comment]
@pauline2k @lyttam - When you have a moment...

File: scripts/db/schema.sql [hunk 1]
```diff
@@ -0,0 +1,168 @@
...
+    id SERIAL PRIMARY KEY,
+    last_guest_user_sync TIMESTAMP WITH TIME ZONE,
+    latest_term_enrollment_csv_set TIMESTAMP WITH TIME ZONE,
+    last_enrollment_sync TIMESTAMP WITH TIME ZONE,
+    last_instructor_sync TIMESTAMP WITH TIME ZONE
+);
+
+--
+
+CREATE TABLE oauth2_data (
```
Comment (pauline2k) [inline scripts/db/schema.sql]: @johncrossman I'm pretty sure we can get by without this oauth2_data table as well. At this point Junction only uses it to upload CSVs to bDrive and we're going to get Ripley out of that business.

File: scripts/db/schema.sql [hunk 2]
```diff
@@ -0,0 +1,168 @@
...
+CREATE TABLE schema_migrations_backup (
+    version CHARACTER VARYING(255)
+);
+
+--
+
+CREATE TABLE schema_migrations_fixed_backup (
+    version CHARACTER VARYING(255)
+);
+
```
Comment (pauline2k) [inline scripts/db/schema.sql]: Are these schema_migrations tables a Rails-specific thing? If so we probably wouldn't be using them in the Ripley world.

File: scripts/db/schema.sql [hunk 3]
```diff
@@ -0,0 +1,168 @@
...
+CREATE TABLE saved_uids (
+    id SERIAL PRIMARY KEY,
+    owner_id CHARACTER VARYING(255),
+    stored_uid CHARACTER VARYING(255),
+    created_at TIMESTAMP WITH TIME ZONE NOT NULL,
+    updated_at TIMESTAMP WITH TIME ZONE NOT NULL
+);
+
+CREATE INDEX saved_uids_owner_id_idx ON saved_uids USING btree (owner_id);
+
```
Comment (pauline2k) [inline scripts/db/schema.sql]: @johncrossman Our snap decision to do without rebuilding view-as logic in Ripley means that we can get by without these recent_uids and saved_uids tables too.

File: scripts/db/schema.sql [hunk 4]
```diff
@@ -0,0 +1,168 @@
...
+);
+
+--
+
+CREATE TABLE user_auths (
+    id SERIAL PRIMARY KEY,
+    uid CHARACTER VARYING(255) NOT NULL,
+    is_superuser BOOLEAN DEFAULT FALSE NOT NULL,
+    active BOOLEAN DEFAULT FALSE NOT NULL,
+    is_viewer BOOLEAN DEFAULT FALSE NOT NULL,
```
Comment (pauline2k) [inline scripts/db/schema.sql]: No view-as also means no need for this is_viewer column.

**Comment by johncrossman** [comment]
@pauline2k - PR amended.

**Comment by pauline2k** [comment]
Thanks @johncrossman!

**Pull request merged**